examples/io_test/io_visu.f90

```diff
@@ -137,17 +139,13 @@ subroutine write_data()
 
       ! Standard I/O pattern - file per field
 #ifdef ADIOS2
-      call decomp_2d_open_io(io_name, "out", decomp_2d_write_mode)
-      call decomp_2d_start_io(io_name, "out")
+      call io%open_start("out", decomp_2d_write_mode)
```

Comment:
Thinking about this (and I realise this reflects the current design) - do you think we should make the single/multi-field mode part of the I/O variable? That way we could eliminate these ADIOS2-specific calls. The user interface would look something like

```
output_dir = "out" ! Just for clarity
call io%open(output_dir, mode=decomp_2d_singlefield)
call decomp_2d_write_one(1, u1, "u1.dat", opt_io=io) ! Writes u1 into out/u1.dat
call decomp_2d_write_one(2, u2, "u2.dat", opt_io=io) ! Writes u2 into out/u2.dat
...
call io%close()
```

and similarly for the multi-field/single file case.

Happy to try an implementation if you think it sounds good?

Reply:
As a side remark, the muli-field mode is specific to MPI. Thus, we could have `decomp_2d_io_write_one` that does the single field IO. And `decomp_2d_io_mpi_write_var` that does the multiple fields in one file.